app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,42 @@
+class ReviewsController < ApplicationController
+  before_action :authorize_user, except: [:index, :show]
+
+  def create
+    @theater = Theater.find(params[:theater_id])
+    @user = current_user
+    @review = Review.new(review_params)
+    @review.theater = @theater
+    @review.user = @user
+
+    if @review.save
+      flash[:notice] = "Review added successfully"
+      redirect_to theater_path(@theater)
+    else
+      flash[:notice] = @review.errors.full_messages
+      redirect_to theater_path(@theater)
```

Comment:
This however should be a re-render of the current page. Otherwise you will lose all form data.